modules.cmake

```diff
@@ -106,5 +106,24 @@ function(add_module_library name)
         DEPENDS ${pcm})
     endforeach ()
   endif ()
+
+  # Common: add sources
   target_sources(${name} PRIVATE ${sources})
+
+  # MSVC
+  if (MSVC)
+    foreach (src ${sources})
+      # compile file as interface
+      set_source_files_properties(${src} PROPERTIES COMPILE_FLAGS /interface)
+
+      # set .ifc file as reference for dependant project
+      get_filename_component(ifc ${src} NAME_WE)
+      set(ifc "${CMAKE_CURRENT_BINARY_DIR}/${ifc}.ifc")
+      target_compile_options(${name} INTERFACE /reference "${ifc}")
+
+      # track generated .ifc file
+      set_target_properties(${name} PROPERTIES ADDITIONAL_CLEAN_FILES ${ifc})
+      set_source_files_properties(${ifc} PROPERTIES GENERATED ON)
```

Comment:
I think we also need a file-level dependency so that when .ifc is removed for some reason it gets rebuilt.

Reply:
I can't believe it, but there seems to be no other way but to use `add_custom_command()` to indicate that this compilation run also has a byproduct in the form of an `.ifc` file.

Maybe we can proceed as is and address this problem separately later.

Reply:
Well, it is called `OBJECT_OUTPUTS` but does not work as expected: file removal does not trigger recompilation
https://cmake.org/cmake/help/latest/prop_sf/OBJECT_OUTPUTS.html

Reply:
OK, let's address it later.